spirv_msl.cpp

```diff
@@ -1176,10 +1179,18 @@ void CompilerMSL::emit_instruction(const Instruction &instruction)
 
 	// Images
 
-	// Reads == fetches in Metal
+	// Reads == Fetches in Metal
 	case OpImageRead:
+	{
+		// Mark that this shader reads from this image
+		uint32_t img_id = ops[2];
+		auto *p_var = maybe_get_backing_variable(img_id);
+		if (p_var)
+			unset_decoration(p_var->self, DecorationNonReadable);
```

Comment:
Thanks...but INABIAF. 😏

I deliberately left the `force_recompile` out of `OpImageRead` because in Metal the default (non-attributed) textures are set to `access::sample` which allows both sampling and reading.

In this context, the `force_recompile` is used to enable the output of the access qualifiers... which only needs to be performed if some combination involving write access is required.